pkg/virt-handler/notify-server/server.go

```diff
@@ -150,13 +150,14 @@ func RunServer(virtShareDir string, stopChan chan struct{}, c chan watch.Event,
 	}()
 
 	// wait for either the server to exit or stopChan to signal
-	select {
-	case <-done:
-		log.Log.Info("notify server done")
-	case <-stopChan:
-		grpcServer.Stop()
-		log.Log.Info("notify server stopped")
+	for {
+		select {
+		case <-done:
+			log.Log.Info("notify server done")
+			return nil
+		case <-stopChan:
```

Comment:
Flow, Client sends ERROR, which triggers 
https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-handler/cache/cache.go#L517
This function closes stopChan, but is blocking until the go routine that runs RunServer would finish,
because it wait for 
https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-handler/cache/cache.go#L525

Imo we should protect the handlers with a reader write lock which is synced with 
https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-handler/cache/cache.go#L527
this channel close

i will work on it if its ok please, thanks